samples/Java/azure-servicebus/QueueClientQuickstart/src/main/java/com/microsoft/azure/servicebus/samples/queueclientquickstart/QueueClientQuickstart.java

```diff
@@ -0,0 +1,101 @@
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class QueueClientQuickstart {
+    // Connection String for the namespace can be obtained from the Azure portal under the
+    // 'Shared Access policies' section.
+
+    // connection string can be initialized via environment variable
+    private static String connectionString = System.getenv("SB_SAMPLES_CONNECTIONSTRING");
+    // queue name can be initialiyzed via environment variable
+    private static String queueName = System.getenv("SB_SAMPLES_QUEUENAME");
+    // queue client instance
+    private static IQueueClient queueClient;
+    // message send/receive counters
+    private static int totalToSend = 100;
+    private static AtomicInteger totalReceived = new AtomicInteger(0);
+    // log4j logger 
+    private static Logger logger = Logger.getRootLogger();
+
+    public static void main(String[] args) throws Exception {
+        // Parse and evaluate command line. Exit when there's not
+        // enough configuration information in environment and to 
+        // command line to continue
+        if (!parseCommandLine(args)) {
+            return;
+        }
+
+        logger.info("Starting BasicSendReceiveWithQueueClient sample");
+
+        // Create a QueueClient instance using the connection string builder
+        logger.info("Create queue client.");
+        queueClient = new QueueClient(new ConnectionStringBuilder(connectionString, queueName), ReceiveMode.PEEKLOCK);
+
+        // send messages in a loop
+        for (int i = 0; i < totalToSend; i++) {
+            int currentMessageCounter = i;
+            queueClient.sendAsync(new Message("" + i)).thenRunAsync(() -> {
+                logger.info(String.format("Sent message #%d.", currentMessageCounter));
+            });
+        }
+
+        // register the anonymous message handler which receives/handles the messages. 
+        queueClient.registerMessageHandler(new IMessageHandler() {
+            // callback invoked when the message handler loop has obtained a message
+            public CompletableFuture<Void> onMessageAsync(IMessage message) {
+                // receives message is passed to callback
+                logger.info(String.format("Received message %d with sq#: %d and lock token: %s.",
+                        totalReceived.incrementAndGet(), message.getSequenceNumber(), message.getLockToken()));
+                return CompletableFuture.completedFuture(null);
+            }
+
+            // callback invoked when the message handler has an exception to report
+            public void notifyException(Throwable throwable, ExceptionPhase exceptionPhase) {
+                logger.error(exceptionPhase + "-" + throwable.getMessage());
+            }
+        },
+                // 1 concurrent call, messages are auto-completed, auto-renew duration
+                new MessageHandlerOptions(1, true, Duration.ofMinutes(1)));
```

Comment:
>true, Duration.ofMinutes(1) [](start = 45, length = 27)

Maybe we should add a comment here for each of the args?

Reply:
there is a comment for each of the args :)